anatomy_of_variables/Anatomy of Variables.md

```diff
@@ -0,0 +1,229 @@
+```
+
+**Global Variables**
+
+These are variables that can be used by two or more methods or function. These variables are assigned outside the methods or functions and they can be manipulated within methods or functions.
+
+The example below shows how global variables are assigned and used in Ruby and JavaScript.
+
+```ruby
+@count = 0
+
+def counter
+    while @count <= 10
+        puts @count
+        @count += 1
+    end
+end
+
+counter #=>0 1 2 3 4 5 6 7 8 9 10
+```
+
+
+```JavaScript
+var count = 0;
+
+function counter(count){
+    while (count <= 10) {
+       console.log(count);
+        count ++;
+    }
+}
+
+counter(); //=>0 1 2 3 4 5 6 7 8 9 10
+```
+
+You will notice that `count` was assigned outside the function or method, and it was incremented within the function or method. Also, we need not pass `count` as an argument when calling the method or function.
+
+
+Based on these two types of variables, Ruby has other types of variables. They are:
+
+* Instance Variables
+* Class Variables
+* Class Instant Variables
+* Constants
+
+**Instance Variables**
+
+Instance Variables are global variables that are **ONLY** used in **Instance Methods** within a class. Instance variables in ruby start with a single  `@` sign. Any variable assigned starting with a single `@` sign can be manipulated within an instance method. Just in case, you don't know what an instance method is in ruby, let's take a quick look.
+
+Ruby instance methods are defined with the keyword `def`, then followed by the name of the method and ended with the keyword `end`. Basically, the common methods in ruby are instance methods. These methods can only be called on an instance of the class they belong to.
+The example below shows an instance variable been used in an instance method.
+
+```ruby
+class User
+    def initialize
+        @name = "Adebayo"
+    end
+
+    def greet
+        puts "Hello #{@name}"
+    end
+end
+
+User.new.greet #=> Hello Adebayo
+```
+
+The above example has instance variable `@name` which holds a name `Adebayo`. The instance method `greet` adds `Hello` to the value of `@name`.  This tells us that instance variables act as global variables **within** a class.  
+
+**Class Variables**
+
+Class variables work exactly like instance methods. These varibales are denoted with a double `@` sign, i.e `@@`. These variables are  **ONLY** used in **Class Methods** within a class. Just in case, you don't know what a class method is in ruby, let's take a quick look.
+
+The keyword `self` makes a class method quite diferent from an instance method. A class method is also defined by the `def` keyword, then the keyword `self` is prpended to the method name. For example, if we were to rewrite the above example as a class method, the method `greet` will be written as `self.greet`. Class methods, however, can be called directly on the class they belong to. The example below shows a class variable been used in a class method.
```

Comment:
prpended => prepended. 
Can you rephrase the use of self in this context. `self` can mean different things in different context. Check out [this blog post](https://www.jimmycuadra.com/posts/self-in-ruby/) to learn more about self. I will also be writing on self in the nearest future.